### PR TITLE
fix: expand tilde in --env-file paths to user home directory

### PR DESCRIPTION
**Pull request by tensorworkerr**
When using `--env-file=/~.env`, the tilde was not expanded to the user's home directory. Instead, it was treated as a literal character and resolved relative to the current working directory, resulting in errors like "couldn't find env `file: /current/dir/~/.env`".

This adds an ExpandUser function that expands ~ to the home directory before converting relative paths to absolute paths.
Fixes https://github.com/docker/compose/issues/13508

What I did
-Added ExpandUser() function in internal/paths/paths.go that expands a leading ~ to the user's home directory using os.UserHomeDir()
-Updated the env file path processing in cmd/compose/compose.go to call ExpandUser() before checking if paths are absolute
-Added unit tests for the new function covering various edge cases (empty string, ~ only, ~/.env, ~/subdir/.env, absolute paths, relative paths, tilde in middle of path, ~otheruser/... syntax)

Related issue
Fixes https://github.com/docker/compose/issues/13508

Contribution by Gittensor, see my contribution statistics at https://gittensor.io/miners/details?githubId=178221229

### CODE REVIEW

File: internal/paths/paths.go [hunk 1]
```diff
@@ -22,6 +22,30 @@ import (
 	"strings"
 )
 
+// ExpandUser expands a leading tilde (~) in a path to the user's home directory.
+// If the path doesn't start with ~, it is returned unchanged.
+// If the home directory cannot be determined, the original path is returned.
+func ExpandUser(path string) string {
```
Comment (ndeloof) [inline internal/paths/paths.go]: you can use `paths.ExpandUser` from compose-go

**Comment by ndeloof** [comment]
A better place to fix this is `GetEnvFromFile` in compose-go: [relativePathsResolver](https://github.com/compose-spec/compose-go/blob/9642b85460f2de66e6295c8b1f2e968fc9c3cbae/paths/resolve.go#L100) already manages this for usage within a compose file as file attributes, would make sense to have this logic all in one place

File: internal/paths/paths.go (hunk 1 above)
Comment (tensorworkerr) [reply internal/paths/paths.go]: Updated as you suggested.

**Comment by thaJeztah** [comment]
@ndeloof I don't think this was a bug, and actually the correct behavior; expansion is handled by the shell, and does not happen if the value is prefixed by a `=`; you'll see the same for any tool, e.g. trying with `git`;

```bash
git --git-dir=~/go/src/github.com/docker/compose/.git describe
fatal: not a git repository: '~/go/src/github.com/docker/compose/.git'

git --git-dir ~/go/src/github.com/docker/compose/.git describe
v2.26.1-891-gb760afaf9
```

Bit concerned that this could be a sliding slope.

**Comment by ndeloof** [comment]
@thaJeztah yes I know shell is in charge for this, but as long as we support both space and equal flag values, need to get an homogeneous behavior. compose.yaml already offers this feature

**Comment by thaJeztah** [comment]
Yeah, the issue here would be that it's not homogeneous across the CLI; other commands will _not_ expand these, so now `docker <some command> --some-flag=~/foo` won't work, but `docker compose --some-flag=~/foo` will.
